// Definitions for common cryptographic enum types.
syntax = "proto3";

package cloud.crypto.tink;

option java_package = "com.google.cloud.crypto.tink";

enum EllipticCurve {
  UNKNOWN_CURVE = 0;
  NIST_P224 = 1;
  NIST_P256 = 2;
  NIST_P384 = 3;
  NIST_P521 = 4;
};

enum HashType {
  UNKNOWN_HASH = 0;
  SHA1 = 1; // Using SHA1 for digital signature is deprecated but HMAC-SHA1 is fine.
  SHA224 = 2;
  SHA256 = 3;
  SHA512 = 4;
};

